core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotationUtil.java

```diff
@@ -64,6 +67,18 @@ enum FindOption {
          * {@link Class}. Otherwise, this option will be ignored.
          */
         LOOKUP_SUPER_CLASSES,
```

Comment:
The search target is coupled with the given `AnnotatedElement`.
It seems clear what the supreme element of the element indicates.

My understanding is that we can merge the three options into `LOOKUP_SUPER_ELELMENTS`.
The behavior will be determined depending on the type of `AnnotatedElement`.
What do you think of that?

Reply:
Oh, thanks. Actually I had considered merging the three options.
However I thought the goal of this PR is to find the Annotation in Method or Parameter in the super class only when it is `@Description` annotation. For other annotation, I thought we need to keep currnet logic, which is not to try to find the Annotation in the super class. Therefore, strictly speaking the search target is not coupled with the given `AnnotatedElement`.

Here is example,
- `AnnotatedElement` = Method && LOOKUP_SUPER_CLASSES && `@Path` => Return annotation only at specific class (without super class)

If we merge three options (=LOOKUP_SUPER_ELEMENTS)
- `AnnotatedElement` = Method && LOOKUP_SUPER_ELEMENTS && `@Path` => Return annotations at all super class

In order to find Annotation in the all super class only when the target annotation is `@Description` and the `AnnotatedElement` is Method or Parameter (or Class), I choose to add two `FindOption` explicitly.

Reply:
I didn't think of the corner case. Let's keep it as is.